File: 198.打家劫舍.py
#
# @lc app=leetcode.cn id=198 lang=python3
#
# [198] 打家劫舍
#

# @lc code=start
class Solution:
    def rob(self, nums: List[int]) -> int:
        max1 = 0
        max2 = 0
        for i in range(len(nums)):
            max3 = max1 + nums[i]
            max1 = max2
            max2 = max(max2, max3)
        return max2
# @lc code=end

File: 198.打家劫舍.cpp
/*
 * @lc app=leetcode.cn id=198 lang=cpp
 *
 * [198] 打家劫舍
 */

// @lc code=start
class Solution {
public:
    int rob(vector<int>& nums) {
        int a = 0, b = 0;
        for (int num: nums) {
            int c = max(b, num + a);
            a = b;
            b = c;
        }
        return b;
    }
};
// @lc code=end

